src/pages/smart-contracts/principals.md

```diff
@@ -102,13 +102,16 @@ For example, a smart contract that implements something like a "token
 faucet" could be implemented as so:
 
 ```clarity
+(define-map claimed-before
+ ((sender principal))
+ ((claimed bool)))
+
 (define-public (claim-from-faucet)
-  (if (is-none (map-get? claimed-before (tuple (sender tx-sender))))
-      (let ((requester tx-sender)) ;; set a local variable requester = tx-sender
-        (begin
-            (map-set claimed-before { sender: requester, claimed: true })
-            (as-contract (stx-transfer? u1 tx-sender requester))))
-      (err 1)))
+  (if (is-none (map-get? claimed-before {sender: tx-sender}))
+    (let ((requester tx-sender))
+      (map-set claimed-before {sender: requester} {claimed: true})
+      (as-contract (stx-transfer? u1 tx-sender requester)))
```

Comment:
Thank you @adamdunkels, apologies for latency on this review!
Instead of `if` I think I would use the pattern "assert / mutate", and clearly explicit the fact that  an error occurring during `stx-transfer?` is being cascaded (it's the last statement), vs being shallowed.
```
(define-map claimed-before
  ((sender principal))
  ((claimed bool)))

(define-constant err-already-claimed u1)
(define-constant err-faucet-empty u2)

(define-public (claim-from-faucet)
    (let ((requester tx-sender) (amount u1)) ;; set a local variable requester = tx-sender
        (asserts! (is-none (map-get? claimed-before {sender: requester})) (err err-already-claimed))
        (unwrap! (as-contract (stx-transfer? amount tx-sender requester)) (err err-faucet-empty))
        (map-set claimed-before {sender: requester} {claimed: true})
        (ok amount)))
```

Reply:
@adamdunkels mind integrating these changes into your pr?

Reply:
@aulneau done!

Reply:
Wonderful. Do you think the list explaining the code below needs to be updated to reflect the new syntax changes?